static/canvas.js

```diff
@@ -126,8 +128,27 @@ class Canvas {
         /* Event for moving the mouse. Primarily used for enlarging a box and moving it. */
         canvas.addEventListener('mousemove', function (e) {
             var mouse = myState.getMouse(e);
-            if (!myState.enlargeDirection && !myState.dragging) {
-                return;
+            var mx = mouse.x;
+            var my = mouse.y;
+            var boxes = myState.boxes;
+            var l = boxes.length;
+            var border = false;
+            var box = null;
+
+            /* Checks if mouse is hovering over a border. */
+            for (var i = l - 1; i >= 0; i--) {
+                box = boxes[i]; 
+                border = boxes[i].withinBorder(mx, my);
+                if (border) {
+                    break;
+                }
+            }
+
+
+            if (!myState.enlargeDirection && !myState.dragging && !border) {
+                document.body.style.cursor = 'default';
+            } else if (!myState.enlargeDirection && !myState.dragging && border) {
+                myState.doubleArrow(box, mx, my);
```

Comment:
Since you already found out which border the mouse is hovering over for all of these cases, you can probably simplify the double arrow function extensively (i.e. get rid of all the conditional clutter and use the border you found).